test/cloudtest/test_storaged.py

```diff
@@ -48,6 +48,49 @@ def test_storaged_creation(mz: MaterializeApplication) -> None:
         wait(condition="condition=Ready", resource=storaged)
 
 
+def test_storaged_resizing(mz: MaterializeApplication) -> None:
+    """Test that resizing a given source causes the storaged to be replaced"""
+    mz.testdrive.run_string(
+        dedent(
+            """
+            $ kafka-create-topic topic=test
+
+            $ kafka-ingest format=bytes topic=test
+            ABC
+
+            > CREATE SOURCE source1
+              FROM KAFKA BROKER '${testdrive.kafka-addr}'
+              TOPIC 'testdrive-test-${testdrive.seed}'
+              FORMAT BYTES
+              ENVELOPE NONE;
+            """
+        )
+    )
+    id = mz.environmentd.sql_query(f"SELECT id FROM mz_sources WHERE name = 'source1'")[
+        0
+    ][0]
+    assert id is not None
+    storaged = f"pod/storage-{id}-0"
+
+    wait(condition="condition=Ready", resource=storaged)
+    unaltered = mz.kubectl("get", storaged, "-o", "jsonpath=jsonpath='{.status.startTime}'")
+
+    mz.testdrive.run_string(
+        dedent(
+            """
+            > ALTER SOURCE source1
+              SET (SIZE '16');
+            """
+        ),
+        no_reset=True,
+    )
+
+    wait(condition="condition=Ready", resource=storaged)
+    altered = mz.kubectl("get", storaged, "-o", "jsonpath=jsonpath='{.status.startTime}'")
+
+    assert unaltered != altered, "Altering a source should restart the backing container"
```

Comment:
This check is a good one but unfortunately it does not guarantee that the pod has been restarted with the desired new command line option of `--workers 16` . I think the kubernetes API should be used to confirm that.

In `test_antiaffinity.py`, in `zones_used`, you can see how the API can be used to query a particular pod. In your case, you need to drill down from spec into `containers` (https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1PodSpec.md) and fish for the command line from there. Maybe the same 

Or you can use your `kubectl` approach:

```
kubectl get pod/storage-u1-0 -o  "jsonpath=jsonpath='{.spec.containers[0].args[0]}'"
```

This currently returns `--workers=1` for me.